src/CommunityToolkit.Maui/Essentials/FileFolderDialog.android.cs

```diff
@@ -0,0 +1,397 @@
+							directory += "/" + newDirName;
+							UpdateDirectory(listItems);
+						}
+						else
+						{
+							Toast.MakeText(context,
+									"Failed to create '" + newDirName + "' folder",
+									ToastLength.Short)
+								?.Show();
+						}
+					})
+					?.SetNegativeButton("Cancel", (_, _) => { })
+					?.Show();
+			};
+			titleLayout1.AddView(newDirButton);
+		}
+
+		// Create View with folder path and entry text box
+		var titleLayout = new LinearLayout(context)
+		{
+			Orientation = Orientation.Vertical
+		};
+
+
+		var currentSelection = new TextView(context)
+		{
+			LayoutParameters = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.WrapContent),
+			Gravity = GravityFlags.CenterVertical,
+			Text = "Current selection:"
+		};
+		currentSelection.SetTextColor(Color.Black);
+		currentSelection.SetTextSize(ComplexUnitType.Dip, 12);
+		currentSelection.SetTypeface(null, TypefaceStyle.Bold);
+
+		titleLayout.AddView(currentSelection);
+
+		titleView = new TextView(context)
+		{
+			Gravity = GravityFlags.CenterVertical,
+			Text = title,
+			LayoutParameters = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.WrapContent)
+		};
+		titleView.SetTextColor(Color.Black);
+		titleView.SetTextSize(ComplexUnitType.Dip, 10);
+		titleView.SetTypeface(null, TypefaceStyle.Normal);
+
+		titleLayout.AddView(titleView);
+
+		if (fileSelectionMode is FileSelectionMode.FileOpen
+									or FileSelectionMode.FileOpenRoot
+									or FileSelectionMode.FileSave
+									or FileSelectionMode.FileSaveRoot)
+		{
+			inputText = new EditText(context)
+			{
+				Text = defaultFileName
+			};
+
+			titleLayout.AddView(inputText);
+		}
+
+		// Set Views and Finish Dialog builder
+		dialogBuilder.SetView(titleLayout);
+		dialogBuilder.SetCustomTitle(titleLayout1);
+		listAdapter = CreateListAdapter(listItems);
+		dialogBuilder.SetSingleChoiceItems(listAdapter, -1, onClickListener);
+		dialogBuilder.SetCancelable(true);
+
+		return dialogBuilder;
+	}
+
+	void UpdateDirectory(in List<string> subDirectories)
+	{
+		subDirectories.Clear();
+		subDirectories.AddRange(GetDirectories(directory));
+
+		if (titleView is not null)
+		{
+			titleView.Text = directory;
+		}
+
+		if (directoriesDialog?.ListView is not null)
+		{
+			directoriesDialog.ListView.Adapter = null;
+			directoriesDialog.ListView.Adapter = CreateListAdapter(subDirectories);
+		}
+
+		if (inputText is not null
+			&& fileSelectionMode is FileSelectionMode.FileOpen
+									or FileSelectionMode.FileOpenRoot
+									or FileSelectionMode.FileSave
+									or FileSelectionMode.FileSaveRoot)
+		{
+			inputText.Text = selectedFileName;
+		}
+	}
+
+	ArrayAdapter<string> CreateListAdapter(List<string>? items)
+	{
+		var adapter = new SimpleArrayAdapter(context ?? throw new InvalidOperationException(),
+			Android.Resource.Layout.SelectDialogItem,
+			Android.Resource.Id.Text1, items ?? Array.Empty<string>().ToList());
```

Comment:
I would like to remove the `.ToList()`, I can see two solutions, one is changing the argument to be `IList` as you can see in the snippet below
```suggestion
	ArrayAdapter<string> CreateListAdapter(IList<string>? items)
	{
		var adapter = new SimpleArrayAdapter(context ?? throw new InvalidOperationException(),
			Android.Resource.Layout.SelectDialogItem,
			Android.Resource.Id.Text1, items ?? Array.Empty<string>());
```

and other solution will be to create a static empty List<string> and use that instead. That way we don't allocate any memory.
```
|         Method |       Mean |     Error |    StdDev | Ratio |   Gen0 | Allocated | Alloc Ratio |
|--------------- |-----------:|----------:|----------:|------:|-------:|----------:|------------:|
|    WithNoCache | 17.2207 ns | 0.0703 ns | 0.0587 ns |  1.00 | 0.0051 |      32 B |        1.00 |
|  WithCacheList |  0.5017 ns | 0.0142 ns | 0.0132 ns |  0.03 |      - |         - |        0.00 |
| WithCacheArray |  0.5133 ns | 0.0098 ns | 0.0092 ns |  0.03 |      - |         - |        0.00 |
```